### PR TITLE
Feature/cache

### PR DESCRIPTION
**Pull request by carlosvillu**
Review!

---
This change is [Reviewable](https://reviewable.io/reviews/carlosvillu/cv-decorators/112)

### CODE REVIEW

**Comment by carlosvillu** [comment]
👍

File: test/cacheSpec.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import {expect} from 'chai'
+
+import cache from '../src/decorators/cache'
+
+describe('Cache', () => {
+  it('Should exist', () => {
+    expect(cache).to.be.a('function')
+  })
+
+  describe('should be able decorate severals class with diferents TTLS', () => {
```
Comment (FranDelaCasa) [inline test/cacheSpec.js]: should be able to decorate several classes with different TTLs ;)

File: test/cacheSpec.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    let clock = null
+    beforeEach(() => clock = sinon.useFakeTimers())
+    afterEach(() => clock.restore())
+
+    it('return same value for long TTL and diferent for shor TTL', () => {
```
Comment (FranDelaCasa) [inline test/cacheSpec.js]: return same value for long TTL and different for short TTL

File: test/cacheSpec.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      const secondFooCall = foo.syncRndNumber()
+      const secondBarCall = bar.syncRndNumber()
+
+      expect(firstFooCall).to.be.not.eql(secondFooCall)
+      expect(firstBarCall).to.be.eql(secondBarCall)
+    })
+  })
+
+  describe('should decorate an async method', () => {
+    it('return twice times the same random numbre without params', (done) => {
```
Comment (FranDelaCasa) [inline test/cacheSpec.js]: return twice the same random number without params

File: test/cacheSpec.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      }
+      const dummy = new Dummy()
+      Promise.all([dummy.asyncRndNumber(), dummy.asyncRndNumber()])
+        .then(([firstCall, secondCall]) => {
+          expect(firstCall).to.be.eql(secondCall)
+          done()
+        })
+    })
+
+    it('return differents numbers if the promise fail', (done) => {
```
Comment (FranDelaCasa) [inline test/cacheSpec.js]: return different numbers if the promise fails

File: test/cacheSpec.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    it('return twice time the same random number without params', () => {
+      class Dummy {
+        @cache()
+        syncRndNumber (num) { return Math.random() }
+      }
+      const dummy = new Dummy()
+      expect(dummy.syncRndNumber()).to.be.eql(dummy.syncRndNumber())
+    })
+
+    it('return differents numbers with diferrents params', () => {
```
Comment (FranDelaCasa) [inline test/cacheSpec.js]: return different numbers with different params

File: test/cacheSpec.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+        class Dummy {
+          @cache() // 500ms by default
+          syncRndNumber (num) { return Math.random() }
+        }
+        const dummy = new Dummy()
+        const firstCall = dummy.syncRndNumber(123)
+        clock.tick(600)
+        expect(dummy.syncRndNumber(123)).to.be.not.eql(firstCall)
+      })
+      it('remaing the cache before ttl ms', () => {
```
Comment (FranDelaCasa) [inline test/cacheSpec.js]: remain

File: test/cacheSpec.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+        class Dummy {
+          @cache() // 500ms by default
+          syncRndNumber (num) { return Math.random() }
+        }
+        const dummy = new Dummy()
+        const firstCall = dummy.syncRndNumber(1234)
+        clock.tick(400)
+        expect(dummy.syncRndNumber(1234)).to.be.eql(firstCall)
+      })
+      it('remaing the cache before not default ttl ms', () => {
```
Comment (FranDelaCasa) [inline test/cacheSpec.js]: remain

**Comment by miduga** [comment]
👍  mola! :)

**Pull request merged**